### PR TITLE
Colab pytorch mnist 

### PR DESCRIPTION
**Pull request by Le-Zheng**

### CODE REVIEW

File: docs/docs/Orca/orca-pytorch-quickstart.md [hunk 1]
```diff
@@ -3,7 +3,9 @@
 
 ### **Step 0: Prepare Environment**
 
-We recommend you to use [Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) to prepare the environment. Please refer to the [install guide](../PythonUserGuide/install/) for more details.
+<a target="_blank" href="https://colab.research.google.com/github/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb"><img src="https://www.tensorflow.org/images/colab_logo_32px.png" />Run in Google Colab</a>&nbsp; <a target="_blank" href="https://github.com/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb"><img src="https://www.tensorflow.org/images/GitHub-Mark-32px.png" />View source on GitHub</a>
```
Comment (jason-dai) [inline docs/docs/Orca/orca-pytorch-quickstart.md:6]: Move this before step 0.

File: docs/docs/Orca/orca-pytorch-quickstart.md [hunk 2]
```diff
@@ -3,7 +3,9 @@
 
 ### **Step 0: Prepare Environment**
 
-We recommend you to use [Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) to prepare the environment. Please refer to the [install guide](../PythonUserGuide/install/) for more details.
+<a target="_blank" href="https://colab.research.google.com/github/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb"><img src="https://www.tensorflow.org/images/colab_logo_32px.png" />Run in Google Colab</a>&nbsp; <a target="_blank" href="https://github.com/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb"><img src="https://www.tensorflow.org/images/GitHub-Mark-32px.png" />View source on GitHub</a>  
+
+We recommend you to use [Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) to prepare the environment. Please refer to the [install guide](../../UserGuide/python.md) for more details.
```
Comment (jason-dai) [inline docs/docs/Orca/orca-pytorch-quickstart.md:8]: [Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) is needed to prepare the Python environment for running this example.

File: readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 3]
```diff
@@ -7,7 +7,16 @@
 
 ### **Step 0: Prepare Environment**
 
-We recommend you to use [Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) to prepare the environment. Please refer to the [install guide](../PythonUserGuide/install/) for more details.
+<table class="tf-notebook-buttons">
```
Comment (jason-dai) [inline readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:10]: move this before step 0.

File: readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 4]
```diff
@@ -7,7 +7,16 @@
...
+  <td>
+    <a target="_blank" href="https://colab.research.google.com/github/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb"><img src="https://www.tensorflow.org/images/colab_logo_32px.png" />Run in Google Colab</a>
+  </td>
+  <td>
+    <a target="_blank" href="https://github.com/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb"><img src="https://www.tensorflow.org/images/GitHub-Mark-32px.png" />View source on GitHub</a>
+  </td>
+</table>
+
+We recommend you to use [Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) to prepare the environment. Please refer to the [install guide](../../UserGuide/python.md) for more details.
```
Comment (jason-dai) [inline readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:19]: Conda is needed to prepare the Python environment for running this example.

File: readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 5]
```diff
@@ -7,7 +7,16 @@
...
 
 **Note:** Conda environment is required to run on the distributed cluster, but not strictly necessary for running on the local machine.
```
Comment (jason-dai) [inline readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:21]: remove this one.

File: docs/docs/Orca/orca-pytorch-quickstart.md [hunk 6]
```diff
@@ -3,7 +3,9 @@
...
 
 **Note:** Conda environment is required to run on the distributed cluster, but not strictly necessary for running on the local machine.
```
Comment (jason-dai) [inline docs/docs/Orca/orca-pytorch-quickstart.md:10]: remove this one

File: docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb [hunk 7]
```diff
@@ -0,0 +1,446 @@
...
+        " "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5C1GYEhGIlWu"
+      },
+      "source": [
+        "[Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) is needed to prepare the Python environment for running this example."
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb:124]: We should mention that the following code block is specific for setting up coda environment on Colab; for general conda installation, the user should refer to https://docs.conda.io/projects/conda/en/latest/user-guide/install/

File: docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb [hunk 8]
```diff
@@ -0,0 +1,446 @@
...
+        "./$MINICONDA_INSTALLER_SCRIPT -b -f -p $MINICONDA_PREFIX\r\n",
+        "\r\n",
+        "# Update Conda\r\n",
+        "conda install --channel defaults conda python=3.6 --yes\r\n",
+        "conda update --channel defaults --all --yes"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb:148]: combine the two above two cells into one

File: docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb [hunk 9]
```diff
@@ -0,0 +1,433 @@
...
+        " "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5C1GYEhGIlWu"
+      },
+      "source": [
+        "[Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) is needed to prepare the Python environment for running this example. The following code cell is specific for setting up conda environment by installing Miniconda on Colab; for general conda installation, please refer to the [install guide](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) for more details."
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb:123]: Put in a separate paragraph:

**Note:** The following code cell is specific for setting up conda environment on Colab; for general conda installation, please refer to the install guide for more details.

File: docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb [hunk 10]
```diff
@@ -0,0 +1,435 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5C1GYEhGIlWu"
+      },
+      "source": [
+        "[Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) is needed to prepare the Python environment for running this example. \r\n",
+        "\r\n",
+        "**Note**: The following code cell is specific for setting up conda environment on Colab; for general conda installation, please refer to the install [guide](https://docs.conda.io/projects/conda/en/latest/user-guide/install/)for more details."
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb:125]: please refer to the [installation guide](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) for more details

File: docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb [hunk 11]
```diff
@@ -0,0 +1,435 @@
...
+        "### **Step 1: Init Orca Context**"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "mXMGhKPk3GYN"
+      },
+      "source": [
+        "OrcaContext.log_output = True # recommended to set it to True when running Analytics Zoo in Jupyter notebook (this will display terminal's  stdout and stderr in the Jupyter notebook).\r\n",
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb:232]: the comment is too long; break it into two lines

File: docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb [hunk 12]
```diff
@@ -0,0 +1,435 @@
...
+      "metadata": {
+        "id": "tQMxn902OBoR"
+      },
+      "source": [
+        "import torch\r\n",
+        "from torchvision import datasets, transforms\r\n",
+        "\r\n",
+        "torch.manual_seed(0)\r\n",
+        "dir='./dataset'\r\n",
+        "batch_size=64\r\n",
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb:332]: can we use batch_size 320?

File: docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb [hunk 13]
```diff
@@ -0,0 +1,435 @@
...
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "6PEmVuGnPFBk"
+      },
+      "source": [
+        "Next, fit and evaluate using the Estimator"
+      ]
+    },
+    {
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb:394]: break fit and evaluate into two blocks

**Comment by jason-dai** [comment]
There is warning messages like:
```
[Stage 7:>                                                          (0 + 1) / 1]Warn: jep.JepException: <class 'StopIteration'>
	at jep.Jep.exec(Native Method)
	at jep.Jep.exec(Jep.java:478)
	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply$mcV$sp(PythonInterpreter.scala:108)
	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply(PythonInterpreter.scala:107)
	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply(PythonInterpreter.scala:107)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

@qiuxin2012

**Comment by qiuxin2012** [comment]
> There is warning messages like:
> 
> ```
> [Stage 7:>                                                          (0 + 1) / 1]Warn: jep.JepException: <class 'StopIteration'>
> 	at jep.Jep.exec(Native Method)
> 	at jep.Jep.exec(Jep.java:478)
> 	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply$mcV$sp(PythonInterpreter.scala:108)
> 	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply(PythonInterpreter.scala:107)
> 	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply(PythonInterpreter.scala:107)
> 	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
> 	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
> 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
> 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> ```
> 
> @qiuxin2012

yes, as python's iterator doesn't have hasnext method, a StopIteration exception will be throwed when iteration is finished.
Do we need to hide this?

**Comment by jason-dai** [comment]
> > There is warning messages like:
> > ```
> > [Stage 7:>                                                          (0 + 1) / 1]Warn: jep.JepException: <class 'StopIteration'>
> > 	at jep.Jep.exec(Native Method)
> > 	at jep.Jep.exec(Jep.java:478)
> > 	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply$mcV$sp(PythonInterpreter.scala:108)
> > 	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply(PythonInterpreter.scala:107)
> > 	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply(PythonInterpreter.scala:107)
> > 	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
> > 	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
> > 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
> > 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
> > 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> > 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> > 	at java.lang.Thread.run(Thread.java:748)
> > ```
> > 
> > 
> > @qiuxin2012
> 
> yes, as python's iterator doesn't have hasnext method, a StopIteration exception will be throwed when iteration is finished.
> Do we need to hide this?

It's very confusing to the user

**Pull request merged**